### ISSUE TITLE
Using dynomite just for dc replication and NOT sharding in any way ?

### ISSUE THREAD

**Issue by daflip** [Netflix/dynomite#476]
Hi

I just came across dynomite and have been researching it and the docs over the last while. While dynomite seems perfect for sharding data across multiple nodes I haven't found anything which gives me a clear answer on whether all this will work (DC <-> DC replication) without any sharding. In other words I'm looking to replicate a single dataset to ALL nodes. If i use the same token everywhere will this give me what I want or is there some other complications that would cause this to not work the way I want ?

Any help or pointers would be much appreciated!

Thanks
Regards
Jerry

**Comment by daflip** [comment]
So after some experimentation it seems this works if i just setup one node per rack. As far as I can see it's working fine this way. One thing I have noticed is that 'keys *' causes the redis client to just hang. Is that a symptom of me doing something wrong or is it not supported ?

**Comment by ipapapa** [comment]
Answering your first question, yes Dynomite can do what you describe. In regards to your second comment, you should never use `KEYS *` against Redis in production. It locks down Redis until it gives all the keys. You should use SCAN instead. More info: https://redis.io/commands/keys

**Comment by daflip** [comment]
Thanks for the reply! Yes, understood and agreed but the codebase I'm dealing with unfortunately has some use of the keys command so I was wondering if it was supported.  It seems that it actually is, but for me on this install (around 180,000 keys) the connection to dynomite from redis-cli just hangs when I issue the command.  Curiously the very same issue happens me when i perform an smembers command on a set which contains quite a lot of data.   I ran dynomite with a verbosity of 11 and when I issue "keys *" it outputs the send receive/send buffer in the logs in blocks.. the blocks repeat a few times as it reads more and more data and then eventually stalls but before the output of the command completes, the last two lines of the buffer is included in the logs below (note that the actual output of the command should actually include more data than what was listed in the logs so it's like it receiving or receiving stalled in some way) -- for the record this is all (dynomite & redis-cli) on the same machine going via the loopback interface. My reading of it is that I would expected the sendv line to have gotten all the way through the buffer were instead it seems to abort before the entire buffer is sent ? Also on the third line before it looks like it outputs two lines in one .. one part of a buffer line and the other the sendv line.  The @ nonsense prefixing each line is just put there by runit -- you can ignore that ;)

```
@4000000059c126e00c064904 00000310  0a 24 34 33 0d 0a 32 38  35 33 2d 63 6f 75 70 6f   |.$43..2853-coupo|
@4000000059c126e00c064904 00000320  6e 2d 62 6f 78 69 6e 67  2d 72 75 73 74 79 2e 68   |n-boxing-rusty.h|
@4000000059c126e00c0654bc 00000330  75 6e[2017-09-19 14:16:54.201] conn_sendv_data:749 sendv on sd 18 163840 of 2088941 in 128 buffers
@4000000059c126e00c06cdd4 [2017-09-19 14:16:54.201] core_loop:717 Flushing writes on CLIENT 18
@4000000059c126e00c06e544 [2017-09-19 14:16:54.201] event_del_out:181 removing conn 0x17edc80(CLIENT) from active
@4000000059c126e00c07086c [2017-09-19 14:16:54.201] stats_swap:1574 skip swap of current 0x17df708 shadow 0x17df768 as aggregator is busy
@4000000059c126e00c0723c4 [2017-09-19 14:16:54.201] core_process_messages:672 length of C2G_OutQ : 0
```
I'm using redis 3.0.7 with dynomite 0.5.9. 

My config is:
```
dyn_o_mite:
  datacenter: dc
  dyn_listen: 0.0.0.0:8102
  dyn_port: 8102
  dyn_seed_provider: simple_provider
  dyn_seeds:
  - 172.30.4.118:8102:rack3:dc:0
  - 172.30.4.116:8102:rack1:dc:0
  listen: 0.0.0.0:8101
  pem_key_file: /etc/dynomite/key.pem
  rack: rack2
  secure_server_option: datacenter
  servers:
  - 127.0.0.1:6379:1
  timeout: 150000
  tokens: 0
```
Any ideas?

**Comment by daflip** [comment]
FWIW the 'disable flush'  branch fixes the issue I'm having. Is this branch safe to use ? comparing v0.5.8 it seems the flushing stuff inside core_loop of dyn_core.c was added in v0.5.9 and this is what's causing the issue

**Comment by shailesh33** [comment]
Thanks Jerry. I am going to issue a fix to merge disable flush in v0.5.9
since there were a few people who complained about it. I do not have the
bandwidth to take a stab at it. I will make a new release on v0.5.9

**Comment by daflip** [comment]
Thank you! If there's anything I can help with let me know
